clients/rospy/src/rospy/impl/tcpros_base.py

```diff
@@ -100,12 +102,19 @@ def recv_buff(sock, b, buff_size):
     @return: number of bytes read
     @rtype: int
     """
-    d = sock.recv(buff_size)
-    if d:
-        b.write(d)
-        return len(d)
-    else: #bomb out
-        raise TransportTerminated("unable to receive data from sender, check sender's logs for details")
+    # add #2123
+    try:
+        d = sock.recv(buff_size)
+        if d:
+            b.write(d)
+            return len(d)
+        else: #bomb out
+            raise TransportTerminated("unable to receive data from sender, check sender's logs for details")
+    except socket.error as ex:
+        if ex.args[0] in (EAGAIN, EWOULDBLOCK):
+            raise TransportTerminated("unable to receive data from sender, check sender's logs for details")
```

Comment:
if either EAGAIN or EWOULDBLOCK, it would want to retry. but raising exception with this particular message, it will break out the while loop in `receive_loop`.

Reply:
I fixed this bug and had to deal with the logic